tests/test_workers/test_tasks/test_build_regenerate_bundle.py

```diff
@@ -644,15 +652,127 @@ def test_adjust_operator_bundle_already_pinned_by_iib(
+        'amqstreams',
+        {'operators.operatorframework.io.bundle.package.v1': 'amqstreams-cmp'},
+    )
+    mock_gil.return_value = {'name': 'namespace/reponame', 'version': 'rhel-8'}
+    manifests_dir = tmpdir.mkdir('manifests')
+    metadata_dir = tmpdir.mkdir('metadata')
+    csv1 = manifests_dir.join('2.clusterserviceversion.yaml')
+    csv2 = manifests_dir.join('3.clusterserviceversion.yaml')
+
+    # NOTE: The OperatorManifest class is capable of modifying pull specs found in
+    # various locations within the CSV file. Since IIB relies on this class to do
+    # such modifications, this test only verifies that at least one of the locations
+    # is being handled properly. This is to ensure IIB is using OperatorManifest
+    # correctly.
+    csv_template = textwrap.dedent(
+        """\
+        apiVersion: operators.example.com/v1
+        kind: ClusterServiceVersion
+        metadata:
+          name: amqstreams.v1.0.0
+          namespace: placeholder
+          annotations:
+            containerImage: {registry}/{operator}{image}{ref}
+        """
+    )
+    csv_related_images_template = csv_template + textwrap.dedent(
+        """\
+        spec:
+          relatedImages:
+          - name: {related_name}
+            image: {registry}/{operator}{image}{related_ref}
+        """
+    )
+    csv1.write(
+        csv_related_images_template.format(
+            registry='quay.io',
+            ref='@sha256:654321',
+            related_name='image-654321-annotation',
+            related_ref='@sha256:654321',
+            operator='operator',
+            image='/image',
+        )
+    )
+    csv2.write(
+        csv_related_images_template.format(
+            # This registry for the company-marketplace will be replaced based on
+            # worker configuration.
+            registry='registry.access.company.com',
+            ref='@sha256:765432',
+            related_name='operator/image-765432-annotation',
+            related_ref='@sha256:765432',
+            operator='operator',
+            image='/image',
+        )
+    )
+
+    bundle_replacements = {
+        'quay.io/operator/image@sha256:654321': 'quay.io/operator/image@sha256:123456',
+    }
+
+    labels = build_regenerate_bundle._adjust_operator_bundle(
+        str(manifests_dir),
+        str(metadata_dir),
+        1,
+        'company-marketplace',
+        pinned_by_iib=True,
+        bundle_replacements=bundle_replacements,
+    )
+
+    # The com.redhat.iib.pinned label is not explicitly set, but inherited from the original image
+    assert labels == {'operators.operatorframework.io.bundle.package.v1': 'amqstreams-cmp'}
+    assert csv1.read_text('utf-8') == csv_related_images_template.format(
+        registry='quay.io',
+        ref='@sha256:123456',
+        related_name='image-654321-annotation',
+        related_ref='@sha256:123456',
+        operator='namespace/reponame',
+        image='-rhel-8-final',
+    )
```

Comment:
I wanted to call out the discrepancy present in this test case. The related names for annotations have the tag (i.e. "tag-like" part which may be a digest) [integrated into the name](https://github.com/containerbuildsystem/operator-manifest/blob/v0.0.5/operator_manifest/operator.py#L267). This will not be updated with the current logic, but I also don't think that this will be an issue.

Reply:
This shouldn't affect us as the names will still be unique, but I opened an issue for it: https://github.com/containerbuildsystem/operator-manifest/issues/56